[nmstate-1.2] nm sriov: Do not touch SR-IOV if not desired

cathay4t opened the pull request:
We should not create SRIOV settings in NetworkManager if that is not
desired.

Integration test case included.
Manual test been done on Mellanox MT27710(mlx5).

ffmancera reviewed (changes requested):
The test failure seems related to this patch. Could you take a look? Thanks! https://github.com/nmstate/nmstate/actions/runs/3125784555/jobs/5070534496#step:5:2332

cathay4t committed:
Fix cargo clippy

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
The CI failure is caused by duplicate call of `create_new_nm_simple_conn()` with modified data from first call. Good example why rust introduced `as_mut()` and `as_ref()`.

Code fixed.

cathay4t committed:
nm sriov: Do not touch SR-IOV if not desired

We should not create SRIOV settings in NetworkManager if that is not
desired.

Integration test case included.
Manual test been done on Mellanox MT27710(mlx5).

Signed-off-by: Gris Ge <fge@redhat.com>
